### PR TITLE
operator/pkg: unit test utilities

### PR DESCRIPTION
**Pull request by mohamedawnallah**
**Description**

In this commit, we unit test utilities on downloader operations and unpacking tar files.

**What type of PR is this?**

**Which issue(s) this PR fixes**:
Part of #5491.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: operator/pkg/util/util.go [hunk 1]
```diff
@@ -74,19 +74,20 @@ func (d *Downloader) Read(p []byte) (n int, err error) {
 	return
 }
 
+var httpClient = http.Client{
+	Timeout: 60 * time.Second,
+}
```
Comment (XiShanYongYe-Chang) [inline operator/pkg/util/util.go:79]: Hi @mohamedawnallah why change it to a global variable?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
When I run the ut in the windows, it report error:
```log
=== RUN   TestDownloadFile/DownloadFile_FileDownlaoded_
    util_test.go:229: failed to prep before downloading the file, got: failed to create temp file: open C:\Users\C00502~1\AppData\Local\Temp/tempfile-*.txt: The filename, directory name, or volume label syntax is incorrect.
    util_test.go:236: unexpected error, got: failed to download file. url: https://www.example.com/test-file code: 503
    util_test.go:242: failed to verify the actual of download of file: failed   to read file: open C:\Users\C00502~1\AppData\Local\Temp/tempfile-*.txt: The filename, directory name, or volume label syntax is incorrect.
--- FAIL: TestDownloadFile (0.00s)
    --- PASS: TestDownloadFile/DownloadFile_UrlIsNotFound_FailedToGetResponse (0.00s)
    --- PASS: TestDownloadFile/DownloadFile_ServiceIsUnavailable_FailedToReachTheService (0.00s)
    --- FAIL: TestDownloadFile/DownloadFile_FileDownlaoded_ (0.00s)
```

File: operator/pkg/util/util.go (hunk 1 above)
Comment (mohamedawnallah) [reply operator/pkg/util/util.go:79]: I changed `httpClient` to be a global variable in order to be able to mock it in the unit tests here ?
https://github.com/mohamedawnallah/karmada/blob/e5d5c7294202b17ab7024daabe23c70716adf3aa/operator/pkg/util/util_test.go#L164

**Comment by mohamedawnallah** [comment]
> When I run the ut in the windows, it report error:
> 
> ```
> === RUN   TestDownloadFile/DownloadFile_FileDownlaoded_
>     util_test.go:229: failed to prep before downloading the file, got: failed to create temp file: open C:\Users\C00502~1\AppData\Local\Temp/tempfile-*.txt: The filename, directory name, or volume label syntax is incorrect.
>     util_test.go:236: unexpected error, got: failed to download file. url: https://www.example.com/test-file code: 503
>     util_test.go:242: failed to verify the actual of download of file: failed   to read file: open C:\Users\C00502~1\AppData\Local\Temp/tempfile-*.txt: The filename, directory name, or volume label syntax is incorrect.
> --- FAIL: TestDownloadFile (0.00s)
>     --- PASS: TestDownloadFile/DownloadFile_UrlIsNotFound_FailedToGetResponse (0.00s)
>     --- PASS: TestDownloadFile/DownloadFile_ServiceIsUnavailable_FailedToReachTheService (0.00s)
>     --- FAIL: TestDownloadFile/DownloadFile_FileDownlaoded_ (0.00s)
> ```

@XiShanYongYe-Chang Good catch! The current unit testing code assumed it was only targeting the Linux platform, so it used "/" as a separator. I've updated it to use `filepath.Join` for platform independence. That said, I haven't been able to test it on Windows since I don't have access to one. Could you please test it again and let me know? 🙏

**Comment by XiShanYongYe-Chang** [comment]
Okay, let me have a retry.

File: operator/pkg/util/util.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply operator/pkg/util/util.go:79]: okay, I get it.

**Comment by mohamedawnallah** [comment]
I've updated the commit. @XiShanYongYe-Chang, I realized that I forgot to replace one instance where "/" was used. 😁 Please pull the latest commit! 🙏

**Comment by XiShanYongYe-Chang** [comment]
It still failed:
```log
=== RUN   TestDownloadFile/DownloadFile_FileDownlaoded_
    util_test.go:230: failed to prep before downloading the file, got: failed to create temp file: open C:\Users\C00502~1\AppData\Local\Temp\tempfile-*.txt: The filename, directory name, or volume label syntax is incorrect.
    util_test.go:237: unexpected error, got: failed to download file. url: https://www.example.com/test-file code: 503
    util_test.go:243: failed to verify the actual of download of file: failed   to read file: open C:\Users\C00502~1\AppData\Local\Temp\tempfile-*.txt: The filename, directory name, or volume label syntax is incorrect.
--- FAIL: TestDownloadFile (0.00s)
    --- PASS: TestDownloadFile/DownloadFile_UrlIsNotFound_FailedToGetResponse (0.00s)
    --- PASS: TestDownloadFile/DownloadFile_ServiceIsUnavailable_FailedToReachTheService (0.00s)
    --- FAIL: TestDownloadFile/DownloadFile_FileDownlaoded_ (0.00s)
```

**Comment by mohamedawnallah** [comment]
@XiShanYongYe-Chang, I've updated the commit. I believe the issue is primarily with the naming of the file `tempfile-*.txt`, but I'm still not 100% sure. Could you please try it again and let me know how it goes? 🙏

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/approve
Thanks~
The CI is running successful on the windows system.

**Pull request merged**